backend/application/services/queue/consumer.py

```diff
@@ -6,85 +7,89 @@
 from settings.default import MAX_TIMEOUT_SECONDS
 
 
+def do_check_task_status(task_id, task_record, task_listener, repository):
+    if task_record.status == TaskStatus.RUNNING:
+        if task_listener is None:
+            logging.warning(f"no listener for tasks {task_id} - skipping")
+            repository.mark_failed(obj=task_record)
+        else:
+            is_done = task_listener.done()
+            logging.info(
+                f"Task[{task_record.name}: {task_id}] -> It's done? R: {is_done}"
+            )
+            if is_done:
+                repository.mark_done(obj=task_record)
+            elif task_service.check_timeout(obj=task_record):
+                task_listener.cancel()
+                repository.mark_cancelled(obj=task_record)
+
+
+def clear_listeners():
+    """
+    if task_id in listeners:
+        del listeners[task_id]
+    """
+
+from application.services.tasks_definition.wait_5s import wait_5s
+
 async def consumer():
+    logging.info("LOL!!!!")
+    try:
+        cm = await TASK_MAP["wait_5s"](task_id="LOL!")
+        logging.info("in")
+        logging.info(cm.get_name())
+        await sleep(15)
+    except Exception as e:
+        logging.error(e)
+    return
+
     with app.app_context():
-        queue_tasks = task_service.queue_tasks
+        # queue_tasks = task_service.queue_tasks
         repository = task_service.repository
         listeners = {}
-        while True:
-            queued_tasks: Sequence[Message] = queue_tasks.receive()
+        # while True:
+        from application.models.models import Task
+        queued_tasks: Sequence[Message] = [
+            Message(body="6494fe71-7505-4415-bc5a-4d16e855ea02"),
+        ]
 
-            for queued_task in queued_tasks:
-                task_id = queued_task.body
-                task_kind = queued_task.kind
-                task_record = repository.get(pk=task_id)
-                if task_record is None:
-                    logging.warning(f"Record not found: {task_id}")
-                    continue
+        for queued_task in queued_tasks:
+            task_id = queued_task.body
+            task_record = repository.get(pk=task_id)
+            if task_record is None:
+                logging.warning(f"Record not found: {task_id}")
+            else:
                 logging.info(
-                    f"task getting[{task_kind}]: {task_record.name} [{task_record.status.name}]-> {task_id}"
+                    f"task getting {task_record.name} [{task_record.status.name}]-> {task_id}"
                 )
 
-                # check if task is done with his purpose of live, it not can not be deleted
-                queue_message_done = False
                 if task_record.name in TASK_MAP:
-                    if task_kind == MessageKind.task:
-                        repository.mark_running(obj=task_record)
+                    # repository.mark_running(obj=task_record)
+                    async with asyncio.timeout(0.1):
```

Comment:
awaiting with a timeout on a per task inside a loop would effectively slow down the loop.